src/mono/wasm/runtime/invoke-js.ts

```diff
@@ -0,0 +1,316 @@
+    }
+    console.log("covert_result_to_cs C");
+    const covertor = js_to_cs_map.get(mono_type);
+    console.log("covert_result_to_cs D");
+    assert(covertor, `Unknow covertor for type ${mono_type} at ret`);
+    console.log("covert_result_to_cs E");
+    covertor(<any>data, value);
+    console.log("covert_result_to_cs F");
+}
+
+function _marshal_int32_to_cs(target: VoidPtr, value: any) {
+    return setI32(target, value);
+}
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function _marshal_void_to_cs(_target: VoidPtr, _value: any) {
+    return 0;
+}
+
+function _marshal_int32_to_js(source: VoidPtr) {
+    //console.log("_marshal_int32_to_js S" + source);
+    const pRef = getU32(source);
+    //console.log("_marshal_int32_to_js P" + pRef);
+    const value = getI32(pRef);
+    //console.log("_marshal_int32_to_js v" + value);
+    return value;
+}
+
+function _marshal_string_to_cs(target: VoidPtr, value: any) {
+    const pError = js_string_to_mono_string(value)!;
+    <MonoString><any>setU32(target, pError);
+}
+
+function _marshal_string_to_js(source: VoidPtr) {
+    //console.log("_marshal_string_to_js S" + source);
+    const pRef = getU32(source);
+    //console.log("_marshal_string_to_js P" + pRef);
+    const pValue = <MonoString><any>getU32(pRef);
+    //console.log("_marshal_string_to_js P2" + pValue);
+    const value = conv_string(pValue);
+    //console.log("_marshal_string_to_js v" + value);
+    return value;
+}
+
+function _marshal_exception_to_cs(target: VoidPtr, value: any) {
+    let res = "unknown exception";
+    if (value) {
+        res = value.toString();
+        const stack = value.stack;
+        if (stack) {
+            // Some JS runtimes insert the error message at the top of the stack, some don't,
+            //  so normalize it by using the stack as the result if it already contains the error
+            if (stack.startsWith(res))
+                res = stack;
+            else
+                res += "\n" + stack;
+        }
+    }
+
+    const pError = js_string_to_mono_string(res)!;
+    setU32(target, pError);
+}
+
+function _marshal_exception_to_js(source: VoidPtr) {
+    const pValue = <MonoString><any>getU32(source);
+    const jsValue = conv_string(pValue);
+    return new Error(jsValue!);
+}
+
+// this will need counterpart in C# to wrap it as JSObject
+function _marshal_js_object_to_cs(target: VoidPtr, value: any) {
+    const js_handle = mono_wasm_get_js_handle(value)!;
+    setU32(target, <any>js_handle);
+}
+
+// this will need counterpart in C# to take handle from JSObject
+function _marshal_js_object_to_js(source: VoidPtr) {
+    const js_handle = <JSHandle><any>getU32(source);
+    const js_obj = mono_wasm_get_jsobj_from_js_handle(js_handle);
+    return js_obj;
+}
+
+
+// this will need counterpart in C# to get C# Object from GCHandle
+function _marshal_cs_object_to_cs(target: VoidPtr, value: any) {
+    const js_handle = mono_wasm_get_js_handle(value)!;
```

Comment:
What is this actually doing? I can't make sense of it. If it's a C# object why is the value 'any' and why are we getting a js handle for it?

Reply:
This is wrong.

Reply:
now it's fixed